src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,110 @@
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateArchiveCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int Timeout_In_Hours = 10;
+        private const int Delay_In_Milliseconds = 10000; // 10 seconds
+
+        public GenerateArchiveCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-archive")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub Migration API's to generate a migration archive";
+            Description += Environment.NewLine;
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to the source GHES API. For GHES, we expect that --ghes-api-url is passed in as the api endpoint for the hostname of your GHES instance. For example: https://api.myghes.com";
+
+            var ghesApiUrl = new Option<string>("--ghes-api-url")
+            {
+                IsRequired = false
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var sourceRepo = new Option<string>("--source-repo")
+            {
+                IsRequired = true
+            };
+            var noSslVerify = new Option("--no-ssl-verify")
+            {
+                IsRequired = false
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(ghesApiUrl);
+            AddOption(githubSourceOrg);
+            AddOption(sourceRepo);
+            AddOption(noSslVerify);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, string, bool, bool>(Invoke);
+        }
+
+        public async Task Invoke(string ghesApiUrl, string githubSourceOrg, string sourceRepo, bool noSslVerify = false, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            _log.LogInformation("Generating Migration Archives...");
+            _log.LogInformation($"GHES SOURCE ORG: {githubSourceOrg}");
+            _log.LogInformation($"GHES SOURCE REPO: {sourceRepo}");
+
+            if (string.IsNullOrWhiteSpace(ghesApiUrl))
+            {
+                _log.LogInformation("--ghes-api-url not provided, defaulting to https://api.github.com");
+                ghesApiUrl = "https://api.github.com";
+            }
```

Comment:
I would also log ghesApiUrl (when provided)

```suggestion
            }
            else
            {
                _log.LogInformation($"GHES API URL: {ghesApiUrl}");
            }
```

Reply:
updated